cpp/Platform.Disposables/DisposableBase.h

```diff
@@ -26,29 +41,34 @@
             return false;
         }
 
-        static DisposableBase() { std::atexit(OnProcessExit); }
 
-        protected: DisposableBase()
+        public: DisposableBase()
         {
+            std::shared_ptr<DisposableBase> this_ptr = std::shared_ptr<DisposableBase>(this);
```

Comment:
This is a bad and not working solution. If we create an object of this class on the stack, then after exiting the constructor, its destructor will be called, and the program will crash due to `std::terminate`, because we will try to free memory on the stack (invalid operation). If We create an object in the form `std::make_shared<T>()`, then we will have two different counters for the number of pointers, which will lead to the fact that we will try to free memory in the heap twice (invalid operation). 
I suggest doing the following:
1) Additionally inherit this class from `std::enable_shared_from_this`
```cpp
class DisposableBase : public std::enable_shared_from_this, public IDisposable { /*...*/};
```
2) In constructor safely get weak_ptr
```cpp
auto this_weak = weak_from_this();
```